src/actors/hcd/neutral_fueling_actor.jl

```diff
@@ -0,0 +1,413 @@
+            x2 = x(ra[i])
+            a1int += (a11 + a12) * (x2 - x1)
+            a11 = a12
+            x1 = x2
+            if i > nra
+                break
+            end
+        end
+        
+        
+        a12 = a1f(i, r[j])
+        x2 = x(r[j])
+        a1int += (a11 + a12) * (x2 - x1)
+        at1[j] = 0.5 * a1int
+        a11 = a12
+        x1 = x2
+    end
+    return at1
+end
+
+"""
+    neucg(dd::IMAS.dd, par::FUSEparameters__ActorNeutralFueling)
+
+This function calculates K matrix of equation 9 in Burrel 1977
+"""
+function get_K_matrix(atrr01::Matrix{Float64},
+                      vth::Vector{Float64},
+                      r::Vector{Float64},
+                      nr::Int,
+                      theta::Vector{Float64},
+                      nt::Int,
+                      sint::Vector{Float64})
+                  
+    k11 = zeros(nr, nr)
+    wt = theta[2] - theta[1]
+    G0(x) = 2.0/sqrt(π)*QuadGK.quadgk(η-> η.^0 .* exp.(-η.^2 .- x ./ η), 0., Inf)[1]
+
+    for i in 1:nr-1
+        for j in (i + 1):nr-1
+            if i == 1
+                apm = atrr01[j, 1]
+                apovp = apm / vth[j]
+                g0ap = g0(apovp)
+    
+                k11[j, 1] = π * g0ap / r[j]
+                apovp = apm / vth[1]
+                g0ap = g0(apovp)
+                k11[1, j] = π * g0ap / r[j]
+            else
+                sum1p, sum1r, sum1wp, sum1wr= 0.0, 0.0, 0.0, 0.0
+                pgt2 = r[j] * r[j]
+                
+                for k in 1:nt
+                    r0 = r[i] * sint[k]
+                    wtx = wt / sqrt(pgt2 - r0 * r0)
+                    
+                    atplt = atrr01[i, k]
+                    atpgt = atrr01[j, k]
+                    ap = atpgt + atplt
+                    am = atpgt - atplt
+                    
+                    # r = rholt=r(i), p=rhogt=r(j)
+                    apovp = abs(ap / vth[j])
+                    amovp = abs(am / vth[j])
+ 
+                    g0ap = g0(apovp)
+                    g0am = g0(amovp)
+
+                    sum1p += wtx * (g0ap + g0am)
+
+                    
+                    # r = rhogt=r(j), p=rholt=r(i)
+                    apovp = abs(ap / vth[i])
+                    amovp = abs(am / vth[i])
+                    g0ap = g0(apovp)
+                    g0am = g0(amovp)
+
+                    sum1r += wtx * (g0ap + g0am)                    
+                end
+                
+                k11[j, i] = sum1p
+                k11[i, j] = sum1r
+            end
+    
+        end
+    end
+    return k11
+end
+
+"""
+    neucg(dd::IMAS.dd, par::FUSEparameters__ActorNeutralFueling)
+
+This model calculates neutral density source fueling based on the neucg 
+model. K. Burrell,  Journal of Computational Physics 27.1 (1978): 88-102.
+"""
+function neucg(dd::IMAS.dd, par::FUSEparameters__ActorNeutralFueling)
+
+    τp = par.τp
+    T_wall = par.T_wall
+    
+    surface = dd.core_profiles.profiles_1d[].grid.surface
```

Comment:
Doing this repeatedly `dd.core_profiles.profiles_1d[]` is "slow" and harder to read.
Can we use cp1d instead

Reply:
fixed in [c81d239](https://github.com/ProjectTorreyPines/FUSE.jl/pull/705/commits/c81d23986fed9eb80ba8fef4239d8f695039f7cd)